accord-core/src/main/java/accord/primitives/ProgressToken.java

```diff
@@ -0,0 +1,72 @@
+package accord.primitives;
+
+import accord.coordinate.Outcome;
+import accord.local.Status;
+import accord.local.Status.Durability;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A representation of activity on a command, so that peers may monitor a command to ensure it is making progress
+ */
+public class ProgressToken implements Comparable<ProgressToken>, Outcome
+{
+    public static final ProgressToken NONE = new ProgressToken(Durability.NotDurable, Status.NotWitnessed, Ballot.ZERO, false);
```

Comment:
Ballot.ZERO usage as a sentinel means it pulls double duty as both the privileged ballot and as an indication of no value.

Reply:
Sort of. Except that once you have any state whatsoever you inherently have a ballot associated with it, so I'm not sure it is even meaningful to say "no value" for a ballot?